test/Sign.Core.Test/SignerTests.cs

```diff
@@ -102,8 +105,12 @@ public async Task SignAsync_WhenFileIsVsix_Signs()
                 FileInfo file = GetTestAsset(temporaryDirectory, "VsixPackage.vsix");
                 FileInfo outputFile = new(Path.Combine(temporaryDirectory.Directory.FullName, "signed.vsix"));
 
-                await SignAsync(temporaryDirectory, file, outputFile);
+                await SignWithAzureKeyVaultAsync(temporaryDirectory, file, outputFile);
+                await VerifyVsixAsync(outputFile, temporaryDirectory);
+
+                outputFile.Delete();
 
+                await SignWithCertificateManagerAsync(temporaryDirectory, file, outputFile);
```

Comment:
You've got 2 tests in one.  These should be separate tests.  In addition, certificate store signing is not exclusive to VSIX signing.  You could sign a NuGet package with a certificate in certificate store, couldn't you?

I think with proper code reuse, you can run the same tests with either provider and have little to no code duplication.

```C#
[Theory]
[MemberData(nameof(GetSigners))]
public async Task SignAsync_WhenFileIsMsixBundle_Signs(SignAsync signAsync)
{
    using (TemporaryDirectory temporaryDirectory = new(new DirectoryService(Mock.Of<ILogger<IDirectoryService>>())))
    {
        FileInfo file = GetTestAsset(temporaryDirectory, "App1_1.0.0.0_x64.msixbundle");
        FileInfo outputFile = new(Path.Combine(temporaryDirectory.Directory.FullName, "signed.msixbundle"));

        await signAsync(_serviceProvider, _certificatesFixture, temporaryDirectory, file, outputFile);

        await VerifyMsixBundleFile(outputFile, temporaryDirectory);
    }
}

public delegate Task SignAsync(IServiceProvider serviceProvider, CertificatesFixture certificatesFixture, TemporaryDirectory temporaryDirectory, FileInfo file, FileInfo outputFile);

public static IEnumerable<object[]> GetSigners()
{
    yield return new SignAsync[] { SignWithAzureKeyVaultAsync };
    yield return new SignAsync[] { SignWithCertificateManagerAsync };
}
```

For this to work, you need to initialize `_serviceProvider` in the constructor and make `TemporaryDirectory` public.